aea/aea.py

```diff
@@ -432,6 +331,49 @@ def _get_behaviours_tasks(
         :return: dict of callable with period specified
         """
         tasks = {}
+
         for behaviour in self.active_behaviours:
             tasks[behaviour.act_wrapper] = (behaviour.tick_interval, behaviour.start_at)
+
         return tasks
+
+    def get_message_handlers(self) -> List[Tuple[Callable[[Any], None], Callable]]:
+        """
+        Get handlers with message getters.
+
+        :return: List of tuples of callables: handler and coroutine to get a message
+        """
+        return super(AEA, self).get_message_handlers() + [
+            (self.filter.handle_internal_message, self.filter.get_internal_message,),
+        ]
+
+    def exception_handler(self, exception: Exception, function: Callable) -> bool:
```

Comment:
Looks like we can easily move this to the agent loop

Reply:
answered above